core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -192,15 +193,24 @@ public final Object visitClearCommand(InvocationContext ctx, ClearCommand comman
                return;
             }
          }
-         // If this node has topology higher than some of the nodes and the nodes could not respond
-         // with the remote entry, these nodes are blocking the response and therefore we can get only timeouts.
-         // Therefore, if we got here it means that we have lower topology than some other nodes and we can wait
-         // for it in StateTransferInterceptor and retry the read later.
-         // TODO: These situations won't happen as soon as we'll implement 4-phase topology change in ISPN-5021
-         throw new OutdatedTopologyException("Did not get any successful response, got " + responses);
+         throw handleMissingSuccessfulResponse(responses);
       });
    }
 
+   protected static CacheException handleMissingSuccessfulResponse(Map<Address, Response> responses) {
+      if (responses.values().stream().anyMatch(UnsureResponse.class::isInstance)) {
+         // We got only unsure responses, as all nodes that were read-owners at the time when we've sent
+         // the request have progressed to newer topology. However we are guaranteed to have progressed
+         // to a topology at most one older, and can immediately retry.
+         return OutdatedTopologyException.getCachedInstance();
+      } else {
+         // Another instance when we don't get any successful response is when all owners are lost. We'll handle
+         // this later in StateTransferInterceptor, as we have to signal this to PartitionHandlingInterceptor
+         // if that's present.
+         return AllOwnersLostException.INSTANCE;
```

Comment:
Both AOLE and OTE will be always caught by STI, if these aren't, it's a bug in STI.

I think I had a case in `NumOwnersNodeCrashInSequenceTest` where I couldn't handle OTEs in PHI, and I had to create AOLEs - I wouldn't bother with another exception type just to make that more readable for you :)